protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -402,7 +403,7 @@ impl BridgeContractCounterparty for MovementClient {
 			to_bcs_bytes(&hash_lock.0).unwrap(),
 			to_bcs_bytes(&time_lock.0).unwrap(),
 			to_bcs_bytes(&recipient.0).unwrap(),
-			to_bcs_bytes(&amount.0).unwrap(),
+			to_bcs_bytes(&amount.weth()).unwrap(),
```

Comment:
The `Amount` type on Movement shouldn't be `weth()` but `Moveth`, this is why originally I was wanting Amount to have a generic param, so that on Movement side it could just be `u64`, we'll have to solve this just with the enum approach now. 

Probably a method `.moveth()`, even if the inner logic of the method is the same, the semantics are wrong.

Reply:
Think that accessing .0 here would be fine, correct?

Reply:
I mean, yes it would work, its just the semantics / method name I'm referring to. It would be confusing for the reader as they would think its actually weth being used here.